### ISSUE TITLE
Reason of ImageDataGenerator 

### ISSUE THREAD

**Issue by engahmed1190** [kuba-siekierzynski/CarL-CNN#2]
Hi @kuba-siekierzynski ,

i would like to thank you for sharing this code , looks promising and good results till now . 
i just want to discuss with you the reason of   `ImageDataGenerator` . i understand that it increase accuracy but how and have you tried any alternatives , also you fit the training data only and don't do the same on the test data .

```
# In order to process data augmentation, keras' ImageDataGenerator can apply some transformations to images on random
from keras.preprocessing.image import ImageDataGenerator

datagen = ImageDataGenerator(
    # featurewise_center=False,
    # samplewise_center=False,
    # featurewise_std_normalization=True,
    # samplewise_std_normalization=True,
    # zca_whitening=True,
    rotation_range=45,
    width_shift_range=0.2,
    height_shift_range=0.2,
    horizontal_flip=True,
    vertical_flip=True)
```

i will keep playing on this model and try to raise the accuracy on some images that fails as steering wheels ones

![668719079510294](https://user-images.githubusercontent.com/25582838/35281951-ecc916ec-005c-11e8-8d1e-26564bc4f654.jpg)

**Comment by kuba-siekierzynski** [comment]
Hi Ahmed,

the reason I used the image data generator was two-fold:

- I only managed to procure some 20k images of 40 different car brands
(relatively few) and most of them were nicely displayed or graphically
"pure"  logos

- images within one class were still similar to each other, as car
logotypes are poorly generalizable (Mercedes is always a tristar in a
circle)

This is why I needed some noise in the data, distortions, skewed images,
shifted and rotated. This helped a little, but in order for the classifier
to generalize even better I would need a really differentiated dataset,
which is hard to obtain other than manually...

Feel free to experiment, perhaps there is still something to be done in
order not to increase accuracy on this set, but to boost an overall
generalization abilities.

To be honest I tried with manually drawn (by hand) logotypes and the
accuracy was lower than this, some 80-83% at best. This is why drawn images
are so much different than those acquired in the Internet, whether pure or
stylized...

Good luck anyway!

Kuba Siekierzynski

23.01.2018 15:47 "Ahmed Osama" <notifications@github.com> napisał(a):

Hi @kuba-siekierzynski <https://github.com/kuba-siekierzynski> ,

i would like to thank you for sharing this code , looks promising and good
results till now .
i just want to discuss with you the reason of ImageDataGenerator . i
understand that it increase accuracy but how and have you tried any
alternatives , also you fit the training data only and don't do the same on
the test data .

# In order to process data augmentation, keras' ImageDataGenerator can
apply some transformations to images on random
from keras.preprocessing.image import ImageDataGenerator

datagen = ImageDataGenerator(
    # featurewise_center=False,
    # samplewise_center=False,
    # featurewise_std_normalization=True,
    # samplewise_std_normalization=True,
    # zca_whitening=True,
    rotation_range=45,
    width_shift_range=0.2,
    height_shift_range=0.2,
    horizontal_flip=True,
    vertical_flip=True)

i will keep playing on this model and try to raise the accuracy on some
images that fails as steering wheels ones

[image: 668719079510294]
<https://user-images.githubusercontent.com/25582838/35281951-ecc916ec-005c-11e8-8d1e-26564bc4f654.jpg>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kuba-siekierzynski/CarL-CNN/issues/2>, or mute the
thread

**Comment by niteOwLTwO** [comment]
Kuba,

i'm new to both Github and machineLearning so all of this is over my head but i still find this project interesting and worth looking into.

i'm starting to learn what machineLearning is all about and am also learning Python.

it will be sometime before i can contribute anything substantial to this project, if ever, but i'm looking forward to help when i can.

i'm following you because i know you from Sololearn and know you are proficient in programning techniques esp Python.

though i know this should not be the place for this kind of comment please excuse me this one time.

**Comment by kuba-siekierzynski** [comment]
niteOwLTwo,
This is totally the best place for this kind of comment :) Any contribution, whether code-related or just motivational - is warmly welcome here. Feel free to give feedback or a question whenever you want.

**Comment by niteOwLTwO** [comment]
thanks Kuba,
i really appreciate your comment.  It's great to know someone on the inside!
